docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,535 @@
+
+```
+GET /wp/v2/widget-areas/footer
+```
+
+```json
+{
+	"id": "footer",
+	...
+	"content": {
+		"raw": "
+			<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-posts-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-comments-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+		",
+		"rendered": ...,
+	}
+}
+```
+
+When a widget area is updated using the REST API, the storage format is _migrated_ and its `'array_version'` is set to `4`.
+
+After migration, the widget area identifier no longer maps to an array of widget identifiers. Instead, it maps to the ID of a `wp_area` post.
+
+`wp_area` is a new post type that denotes posts that contain block markup for a widget area. The `wp_area` post type is similar to the `wp_block` post type, except that it is marked as private and therefore not viewable by accessing `wp-admin/edit.php?post_type=wp_area`.
+
+Let's illustrate with an example. Say that we update the `'footer'` widget area to include a non-`core/legacy-widget` block.
+
+```
+PUT /wp/v2/widget-areas/footer
+
+{ "content": { "raw": "
+	<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+	<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+	<!-- wp:legacy-widget {\"identifier\":\"recent-posts-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+	<!-- wp:legacy-widget {\"identifier\":\"recent-comments-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+" } }
+```
+
+```json
+{
+	"id": "footer",
+	...
+	"content": {
+		"raw": "
+			<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+			<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-posts-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+			<!-- wp:legacy-widget {\"identifier\":\"recent-comments-2\",\"instance\":{\"title\":\"\",\"number\":5}} /-->
+		",
+		"rendered": ...
+	}
+}
+```
+
+This triggers a migration. `'footer'` now points to a numeric ID instead of an array. Other widget areas are not affected.
+
+```php
+array(
+	'footer' => 123,
+	'sidebar' => array(
+		'recent-posts-2',
+		'recent-comments-2',
+	),
+	'array_version' => 4,
+)
+```
+
+Here, `123` is the ID of a `wp_area` post. The `post_content` field of `get_post( 123 )` contains the `'footer'` widget area's block markup.
+
+```html
+<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->
+<!-- wp:legacy-widget {"identifier":"search-2","instance":{"title":""}} /-->
+<!-- wp:legacy-widget {"identifier":"recent-posts-2","instance":{"title":"","number":5}} /-->
+<!-- wp:legacy-widget {"identifier":"recent-comments-2","instance":{"title":"","number":5}} /-->
+```
+
+## New APIs
+
+### `register_widget_area()`
```

Comment:
These new APIs don't seem crucial for this particular RFC. Should we leave them out and focus on the storage and the REST API instead? I'm not certain yet if these are really needed and I would prefer if we discuss these separately. They can be added later to the same RFC.

Reply:
Yeah, fair enough. Removed in 12d797269.